### PR TITLE
Detail error info

### PR DESCRIPTION
**Pull request by vasiliy-zaznobin**
Introduce macro **Err!**
Replace the following code:
```rust
    error_print!("Caller pubkey: {}", &caller_info.key.to_string());
    return Err(ProgramError::InvalidArgument);
```
To
```rust
    return Err!(ProgramError::InvalidArgument; "Caller pubkey: {} ", &caller_info.key.to_string());
```

Introduce macro **E!**
Replace the following code:
```rust
    map_err(|_| ProgramError::InvalidArgument)
```
To
```rust
    map_err(|s| E!(ProgramError::InvalidArgument; "s={:?}", s))
```

### CODE REVIEW

**Review by vakond — APPROVED** [review]
Brilliant idea with the macros.

File: evm_loader/program/src/error.rs [hunk 1]
```diff
@@ -26,4 +26,34 @@ impl<T> DecodeError<T> for EvmLoaderError {
...
+    solana_program::msg!("{}:{:?}", fl, ln);
+    Err(err)
+}
+
+pub fn err_fn<T:>(err: ProgramError, fl: &str, ln: u32, info: &str) -> Result<T, ProgramError> {
+    solana_program::msg!("{}:{:?} : {}", fl, ln, info);
+    Err(err)
+}
+
+macro_rules! Err {
```
Comment (vakond) [inline evm_loader/program/src/error.rs:41]: Maybe it would make sense to add comments describing details and differencies of macros.

**Comment by s-medvedev** [comment]
@vasiliy-zaznobin Сan you give an example of a log that is written in the recipe for a Solana transaction on error?

**Comment by vasiliy-zaznobin** [comment]
See example in solana.log (search: **.rs**)
https://buildkite.com/cyberway/evm-loader/builds/1123#851a491a-9951-446f-9e85-678faac2cc3a

File: evm_loader/program/src/error.rs (hunk 1 above)
Comment (vasiliy-zaznobin) [reply evm_loader/program/src/error.rs:54]: Fixed

**Review by s-medvedev — APPROVED** [review]

**Pull request merged**